core/lib/libmbedtls/src/tee_lmd_provider.c

```diff
@@ -75,6 +85,96 @@ static int mbd_rand(void *rng_state, unsigned char *output, size_t len)
 	* defined(_CFG_CRYPTO_WITH_DRBG)
 	*/
 
+#if defined(_CFG_CRYPTO_WITH_CIPHER) || defined(_CFG_CRYPTO_WITH_MAC) || \
+	defined(_CFG_CRYPTO_WITH_AUTHENC)
+/*
+ * Get the Mbedtls chipher info given a TEE Algorithm "algo"
+ * Return
+ * - TEE_SUCCESS in case of success,
+ * - NULL in case of error
+ */
+static const mbedtls_cipher_info_t
+		*tee_algo_to_mbedtls_cipher_info(uint32_t algo,
+						size_t key_len)
+{
+	/* Only support key_length is 128 bits of AES in Optee_os */
+	switch (algo) {
+#if defined(CFG_CRYPTO_AES)
+	case TEE_ALG_AES_ECB_NOPAD:
+		if (key_len == 128)
+			return mbedtls_cipher_info_from_string("AES-128-ECB");
```

Comment:
It's a pity that `aes_128_ecb_info` is a static variable or we could have returned `&aes_128_ecb_info` directly instead of going via yet one more lookup function. Speaking of lookup functions, if we now have to use one I'd prefer `mbedtls_cipher_info_from_type()` instead of string based scheme.

Reply:
Ok, `mbedtls_cipher_info_from_type` may be better. I will update it.